colour/volume/macadam_limits.py

```diff
@@ -142,3 +143,197 @@ def is_within_macadam_limits(
+
+    Parameters
+    ----------
+    target_brightness : floating point
+        brightness has to be between 0 and 1
+
+    illuminant: object
+        illuminant must be out of colorimetry.MSDS_CMFS['XXX']
+        If there is no illuminant or it has the wrong form,
+        the illuminant SDS_ILLUMINANTS['E']
+        is chosen which has no influence to the calculations,
+        because it is an equal-energy-spectrum
+
+    if necessary a third parameter for the
+    colour-matching function could easily be implemented
+
+    Returns
+    -------
+    an array of CIE -X,Y,Z - Triples for every single wavelength
+    in single nm - Steps in the range from 360 to 830 nm
+
+    References
+    ----------
+    -   cite: Wyszecki, G., & Stiles, W. S. (2000).
+        In Color Science: Concepts and Methods,
+        Quantitative Data and Formulae (pp. 181–184). Wiley.
+        ISBN:978-0-471-39918-6
+    -   cite: Francisco Martínez-Verdú, Esther Perales,
+        Elisabet Chorro, Dolores de Fez,
+        Valentín Viqueira, and Eduardo Gilabert, "Computation and
+        visualization of the MacAdam limits
+        for any lightness, hue angle, and light source," J.
+        Opt. Soc. Am. A 24, 1501-1515 (2007)
+    -   cite: Kenichiro Masaoka. In OPTICS LETTERS, June 15, 2010
+        / Vol. 35, No. 1 (pp. 2031 - 2033)
+
+    Example
+    --------
+    from matplotlib import pyplot as plt
+    import numpy as np
+    import math
+    fig = plt.figure(figsize=(7,7))
+    ax = fig.add_axes([0,0,1,1])
+    illuminant = colour.SDS_ILLUMINANTS['D65']
+
+    def plot_Narrowband_Spectra (Yxy_Narrowband_Spectra):
+        FirstColumn = 0
+        SecondColumn = 1
+
+        x = Yxy_Narrowband_Spectra[...,FirstColumn]
+        y = Yxy_Narrowband_Spectra[...,SecondColumn]
+        ax.plot(x,y,'orange',label='Spectrum Loci')
+
+        x = [Yxy_Narrowband_Spectra[-1][FirstColumn],
+        Yxy_Narrowband_Spectra[0][FirstColumn]]
+        y = [Yxy_Narrowband_Spectra[-1][SecondColumn],
+        Yxy_Narrowband_Spectra[0][SecondColumn]]
+        ax.plot(x,y,'purple',label='Purple Boundary')
+        return()
+
+    for n in range(1, 20):
+        Yxy_Narrowband_Spectra = colour.XYZ_to_xy(
+        colour.macadam_limits(n/20, illuminant) / 100)
+        plot_Narrowband_Spectra (Yxy_Narrowband_Spectra)
+
+    plt.show()
+    """
+    target_bright = target_brightness
+    if target_bright > 1 or target_bright < 0:
+        raise TypeError(
+            "brightness of function macadam_limits( )"
+            "has to be between 0 and 1"
+        )
+    standard_cfms = MSDS_CMFS["CIE 1931 2 Degree Standard Observer"]
+    X_cie31 = standard_cfms.values[..., 0]
+    Y_cie31 = standard_cfms.values[..., 1]
+    Z_cie31 = standard_cfms.values[..., 2]
+    try:
+        illuminant.interpolator
+    except AttributeError:
+        illuminant = SDS_ILLUMINANTS["E"]
+
+    # If there is no illuminant or it has the wrong form,
```

Comment:
Most of those comments could be removed as they are not providing useful information and add a lot of noise when reading the code.

Reply:
Changed in actual version.